pkg/detectors/ftp/ftp.go

```diff
@@ -80,7 +80,13 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		if verify {
 			timeout := s.verificationTimeout
 			if timeout == 0 {
-				timeout = defaultVerificationTimeout
+				if dl, ok := ctx.Deadline(); ok {
+					// Here we assign the remaining time before our context expires
+					// This help us cater the timelimit set through --detector-timeout flag
+					timeout = time.Until(dl)
+				} else {
+					timeout = defaultVerificationTimeout
+				}
```

Comment:
It looks fine. This is what I suggested [here](https://github.com/trufflesecurity/trufflehog/issues/3714#issuecomment-2565522620), but as per @rgmz comment, ftp.Dial can still hang for a longer time. Could you test it with this change?